### PR TITLE
Print backends and architectures for Kokkos

### PR DESCRIPTION
**Pull request by masterleinad**
In reference to https://github.com/dealii/dealii/pull/14377/files#r1022862702.

### CODE REVIEW

**Review by tjhei — APPROVED** [review]
very nice!

File: cmake/setup_write_config.cmake [hunk 1]
```diff
@@ -225,6 +225,10 @@ FOREACH(_feature ${_deal_ii_features_sorted})
       _detailed("#            TBB_VERSION = ${TBB_VERSION}\n")
     ENDIF()
 
+    IF(_feature MATCHES "KOKKOS")
```
Comment (tjhei) [inline cmake/setup_write_config.cmake:228]: is this also printing info if kokkos is disabled?

**Review by Rombur — APPROVED** [review]

File: cmake/setup_write_config.cmake (hunk 1 above)
Comment (masterleinad) [reply cmake/setup_write_config.cmake:228]: No, it doesn't (but we agreed on making `Kokkos` a required dependency anyway).

Comment (tjhei) [reply cmake/setup_write_config.cmake:228]: oh, right.

**Comment by masterleinad** [comment]
Sample output in `details.log` is
```
[...]
#        DEAL_II_WITH_KOKKOS set up with external dependencies
#            KOKKOS_BACKENDS = THREADS;SERIAL
#            KOKKOS_ARCHITECTURES = A64FX
#            KOKKOS_DIR = /tmp/Software/kokkos-install
#            KOKKOS_CXX_FLAGS = -march=armv8.2-a+sve
#            KOKKOS_LINKER_FLAGS = -DKOKKOS_DEPENDENCE -march=armv8.2-a+sve
#            KOKKOS_INCLUDE_DIRS = /tmp/Software/kokkos-install/include
#            KOKKOS_USER_INCLUDE_DIRS = /tmp/Software/kokkos-install/include
#            KOKKOS_LIBRARIES = /tmp/Software/kokkos-install/lib/libkokkoscore.a;/tmp/Software/kokkos-install/lib/libkokkoscontainers.a
[...]
```

**Review by tamiko — APPROVED** [review]

**Review by tjhei — APPROVED** [review]

**Pull request merged**